### DISCUSSION TITLE
How can I use my projects root .env?

### DISCUSSION THREAD

**Discussion by lucassimines** [nuxt/nuxt#17422] [Questions]
How can I make my Nuxt application read my `.env` located in my Project root?

This is my project structure:

```
admin
....nuxt
server
....server files
.env
```

**Comment by jolo-dev** [comment]
I have the same question.
[Docs](https://v3.nuxtjs.org/guide/features/runtime-config/) are not sufficient.
I also tried the [Vite-approach](https://vitejs.dev/guide/env-and-mode.html#env-files)
In the end, I put `import 'dotenv/config'` on every file which needs the `.env`
But I have the feeling that is not the solution...Frustrating

**Comment by Flanker72** [comment]
Both `.env` and `.env.local` files are handled by default. All defined variables become accessible via `process.env.VAR_NAME`.

However, it's strongly recommended to define [app runtime config](https://v3.nuxtjs.org/guide/features/runtime-config) and use `useRuntimeConfig()` to access variables in more convenient way, regarding the client-side.

To override any runtime config variable there is a hint: all variables starting with `NUXT_` are passed there.
You may reference [my article](https://medium.com/@flanker72/nuxt3-complex-solutions-database-integration-8df941f0fb82) to see how to use them with the recommended working approach.

Consider that runtime config have 2 parts: private and public. Public variables are passed to the client-side, private are only accessible from the server-side.

So, `NUXT_MONGO_URL` will be accessible from the server-side as `useRuntimeConfig().mongoUrl`, and `NUXT_PUBLIC_API_URL` will be accessible from both the server and client sides as `useRuntimeConfig().public.apiUrl`.

**Comment by xjccc** [comment]
now you can use this https://nuxt.com/docs/api/commands/dev
